modules/core/src/dxt.cpp

```diff
@@ -2162,9 +2162,7 @@ static bool ocl_dft(InputArray _src, OutputArray _dst, int flags, int nonzero_ro
     // if output format is not specified
     if (complex_output + real_output == 0)
     {
-        if (real_input)
-            real_output = 1;
-        else
+        if (!real_input)
             complex_output = 1;
```

Comment:
Looks like this code is not equivalent.

Does this warning is "set but not used"?
My suggestion is to drop "real_input" and "real_output" flags (replace them to "!complex_input" and "!complex_output") - we don't support [quaternions](http://ieeexplore.ieee.org/document/615636/) or similar here.

Reply:
I've refactored this part of function to be more readable.